### PR TITLE
V3.x.x

### PR DESCRIPTION
**Pull request by gabrielcsapo**
- updates code to be es6ified
- updates code dependency
- updates all depends
- uses influx line syntax
- uses the new stream interface for good reporters
- convert object to line protocol

### CODE REVIEW

**Comment by shaunwarman** [comment]
I really like the update and it's definitely necessary for influxDB versions `0.9.x` and above. A couple things I see:
- Do we want to go all es6 and add an update to required engine? (I see some es6 items sprinkled in with recent changes.)
- We should probably pull out some of the old tests in `report` section that are half-baked *(my fault)* and conform to the stream updates.
- May need to rebase to clean up some extra commits. (Can do through github 🎉 )

**Comment by gabrielcsapo** [comment]
@shaunwarman

**Comment by gabrielcsapo** [comment]
@shaunwarman thoughts?

File: lib/index.js [hunk 1]
```diff
@@ -1,102 +1,88 @@
...
+        if (settings.errorThreshold === null) {
+            settings.errorThreshold = -Infinity;
+        }
 
-internals.GoodInflux.prototype.stop = function () {
-    this._flush();
-};
+        // Try to parse credentials from url. (Overrides options.)
+        if (typeof url.auth === 'string') {
+            var auth
```
Comment (shaunwarman) [inline lib/index.js]: lingering `var`

**Comment by shaunwarman** [comment]
Looks great, I see some lingering `var`'s and then just this travis-ci issue.
Is in the [environment?](https://docs.travis-ci.com/user/languages/javascript-with-nodejs#Node.js-v4-(or-io.js-v3)-compiler-requirements)

**Comment by shaunwarman** [comment]
👍

**Comment by gabrielcsapo** [comment]
@totherik thoughts?

File: lib/mapper.js [hunk 2]
```diff
@@ -101,22 +95,31 @@ var internals = {
...
 
+function formatField(k, v) {
+    const value = Stringify(v);
+    return value === undefined ? '' : k + '=' + value;
+}
 
-exports.prepare = function map(arr) {
-    var points, i, len, eventData, mapper;
+function formatLines(lines) {
+    return lines.map(function (line) {
```
Comment (totherik) [inline lib/mapper.js]: this can be simplified to:

```js
return lines.map(formatLine).join('\n');
```

File: lib/mapper.js [hunk 3]
```diff
@@ -15,45 +14,40 @@ var internals = {
...
-                    } else {
-                        value = Stringify(value);
+            if (column === 'time') {
+                line['timestamp'] = value;
+            } else {
+                if(typeof value === 'object') {
+                  Object.keys(value).forEach(function(k, v) {
+                    let field = formatField(column + k, v);
+                    if(field) {
+                        line['fields'].push(field);
```
Comment (totherik) [inline lib/mapper.js]: why different notation for referencing properties? wherever you set properties of `line` you use bracket notation, whereas elsewhere (constructor, `formatLine`, etc.) you use dot notation.

Comment (gabrielcsapo) [inline lib/mapper.js]: I added an eslint rule to enforce dot notation

File: lib/mapper.js (hunk 2 above)
Comment (gabrielcsapo) [inline lib/mapper.js]: fixed

File: lib/mapper.js [hunk 4]
```diff
@@ -15,45 +14,40 @@ var internals = {
...
-                        value = String(value);
-                    } else if (value instanceof Date) {
-                        value = String(value);
-                    } else {
-                        value = Stringify(value);
+            if (column === 'time') {
+                line.timestamp = value;
+            } else {
+                if(typeof value === 'object') {
+                  Object.keys(value).forEach(function(k, v) {
```
Comment (totherik) [inline lib/mapper.js]: The argument naming here might confuse readers. The second position of the `forEach` function is the index of the array being iterated over (in this case the array of keys). `v` suggests it's the value, which it's not. If that is intended to be the value for the given key/property, that needs to be explicitly looked up via `value[k]`.

Comment (totherik) [inline lib/mapper.js]: (`Object.entries` would be nice here...)

Comment (gabrielcsapo) [inline lib/mapper.js]: @totherik what if I named then `subK` and `subV` since they are sub key and values?

Comment (totherik) [inline lib/mapper.js]: `v` is not a value, it's the index of the `k` in the keys array. If you want the value for that key (`k`) in the `values` object you need to look it up explicitly. (See the 3rd sentence in my original comment.)

Comment (totherik) [inline lib/mapper.js]: If it's intended to be the index, I'd just call it `index`.

Comment (totherik) [inline lib/mapper.js]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach#Parameters

Comment (gabrielcsapo) [inline lib/mapper.js]: @totherik okay, I renamed the value to index.

**Comment by totherik** [comment]
I _think_ this LGTM. 👍

**Comment by shaunwarman** [comment]
Cool 👍

**Comment by shaunwarman** [comment]
@totherik want to do the honors?
```
swarman$ npm owner ls good-influxdb
totherik <totherik@gmail.com>
```

**Comment by gabrielcsapo** [comment]
WOOOT!

**Pull request merged**